### PR TITLE
skeleton outline for episode 1

### PR DESCRIPTION
**Pull request by colinsauze**
An outline for episode 1 (introduction), including some learning objectives based on issue #42. Suggestions for a couple of exercises on the capabilities of deep learning and preparing data.

Please delete the text below before submitting your contribution. 

---

Thanks for contributing! If this contribution is for instructor training, please send an email to checkout@carpentries.org with a link to this contribution so we can record your progress. You’ve completed your contribution step for instructor checkout just by submitting this contribution.  

Please keep in mind that lesson maintainers are volunteers and it may be some time before they can respond to your contribution. Although not all contributions can be incorporated into the lesson materials, we appreciate your time and effort to improve the curriculum.  If you have any questions about the lesson maintenance process or would like to volunteer your time as a contribution reviewer, please contact Kate Hertweck (k8hertweck@gmail.com).  

---

### CODE REVIEW

**Comment by psteinb** [comment]
@colinsauze Just to reiterate, this would be a prefix lesson before the actual lessons start that are laid out in [_extras/lesson-design.md](/carpentries-incubator/deep-learning-intro/_extras/lesson-design.md)

File: _episodes/01-introduction.md [hunk 1]
```diff
@@ -3,12 +3,202 @@ title: "Introduction"
...
+
+> ## What is DL - solution
+>
+> learning outcome -> it is vague, find your own
+> OR
+> hah! both are right in some ways, explain why NN has its place there as well
+>
+{: .solution}
+
+*the neuron/perceptron/unit/
```
Comment (dafnevk) [inline _episodes/01-introduction.md:107]: I don't think explanation on what neurons / perceptrons / units are should go in this episode. We didn't define it as one of the learning goals, and I believe that the intention of episode 2 is to explain what fully connected layers are, while explaining the concept of a neuron on the go.

File: _episodes/01-introduction.md [hunk 2]
```diff
@@ -3,12 +3,202 @@ title: "Introduction"
...
+
+* inputs, weight, sum, non-linearity, output
+* -> DL: stacked perceptrons
+
+* training challenges
```
Comment (dafnevk) [inline _episodes/01-introduction.md:112]: I also wouldn't mention these points yet. I think it's much easier to explain these concepts 'on the go' if we created an actual network in Keras, which is done in [episode 3](https://github.com/carpentries-incubator/deep-learning-intro/issues/44).

File: _episodes/01-introduction.md [hunk 3]
```diff
@@ -3,12 +3,202 @@ title: "Introduction"
...
+  * gradient
+  * overfitting
+
+## Building blocks of NN
+
+*building blocks of a DL model:
+    *example case CNN
```
Comment (dafnevk) [inline _episodes/01-introduction.md:119]: CNNs are handled in [episode 4](https://github.com/carpentries-incubator/deep-learning-intro/issues/45), I don't think we should mention them here yet.

File: _episodes/01-introduction.md [hunk 4]
```diff
@@ -3,12 +3,202 @@ title: "Introduction"
...
+        *show how each layer may look
+        *eg based on face features
+        *we learn from many examples what and eye looks like and where it is located
+*-> needs high amount of data
+
+## Deep Learning workflow
```
Comment (dafnevk) [inline _episodes/01-introduction.md:125]: I think it would be good to do this first as a (group) exercise, because that reveals nicely what people already know about machine learning workflows.

File: _episodes/01-introduction.md [hunk 5]
```diff
@@ -3,12 +3,202 @@ title: "Introduction"
...
+* many different options
+    * name them (and known limitations)
+
+* why keras (and why 'with' tensor-flow)
+    * easy start
+    * huge community
+        * help and tutorials online
+    *
+
+> ## Keras
```
Comment (dafnevk) [inline _episodes/01-introduction.md:170]: I think this can be skipped, will be done in episode 2.

**Review by dafnevk — COMMENTED** [review]
I realize this is still work in progress, but I made some comments already on the skeleton so far. I think we should be careful to not make this episode too long and do too much 'explaining' before we get to the actual coding in episode 2. But we do need to give enough context of course, so there's a balance there. 

I would say (group) exercises should play an important role in this episode, because many people will already have some ideas on what deep learning can do for them, and it helps the instructor to know about the backgrounds and to tackle misconceptions.

I also think it's important to make as much as possible references to applications of DL in academic research contexts. e.g. the ones describes in our learner profiles. This way, we connect to the needs of our audience and it makes the lesson different from the many existing tutorials online.

**Comment by colinsauze** [comment]
Thanks for the comments @dafnevk. Its worth pointing out that I merged the intro branch by @samumantha into what i've done as they overlapped (and I only noticed it existed after i'd made the initial pull request). I'm hoping to find some time next week to work on this and get it into a complete draft lesson. I'll be sure to follow the suggestions you've made.

File: _episodes/01-introduction.md [hunk 6]
```diff
@@ -3,12 +3,202 @@ title: "Introduction"
...
+
+(one way, open for discussion and to be filled with information)
+1. Formulate/ Outline the problem
+
+2. Identify inputs and outputs
+
+3. Prepare data
+common steps for data preparation
+
+4. Choose a cost function and metrics
```
Comment (bpmweel) [inline _episodes/01-introduction.md:135]: Would it make sense to move this to the training model section?
Using Keras  it usually only appears when you define the training algorithm

File: _episodes/01-introduction.md [hunk 7]
```diff
@@ -3,12 +3,202 @@ title: "Introduction"
...
+
+5. Choose a pretrained model or start building architecture from scratch
+build from scratch vs reuse vs transfer learning
+
+6. Train model
+
+7. Tune hyperparameters
+
+8. 'predict'
```
Comment (bpmweel) [inline _episodes/01-introduction.md:144]: Before, or perhaps instead of, predict I would add a section on measuring performance. This would neatly align the tuning of hyper parameters with a validation set and performance measurement with a test set.
